apps/expo/src/components/view-all-display/tests/ViewAllTestDisplay.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FC } from "react";
+import { SafeAreaView, TouchableOpacity } from "react-native";
+import ViewAllScreenTestCard from "./ViewAllScreenTestCard";
+import ViewAllScreenHeader from "../../headers/ViewAllScreenHeader";
+import { FlashList } from "@shopify/flash-list";
+import { trpc } from "../../../utils/trpc";
+
+interface Props {
+  testsFor: "discover" | "trending" | "topPicks";
+}
+
+export const ViewAllTestDisplay: FC<Props> = (props) => {
+  const { data } = trpc.test.getAll.useQuery();
+  let fetchedData = data;
+  let headerTitle = "";
```

Comment:
kinda problematic to use let since it's mutable, can you use ts-pattern for pattern matching instead? https://github.com/gvergnaud/ts-pattern